app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt

```diff
@@ -55,28 +53,56 @@ class SuggestedEditsCardView(context: Context) : DefaultFeedCardView<SuggestedEd
     }
 
     private fun updateContents() {
-        if (isTranslation) {
-            sourceDescription = sourceSummary!!.description!!.capitalize()
-            viewArticleSubtitle.visibility = View.VISIBLE
-            viewArticleSubtitle.text = sourceDescription
+        viewArticleSubtitle.visibility = View.GONE
+        when (card!!.invokeSource) {
+            FEED_CARD_SUGGESTED_EDITS_TRANSLATE_DESC -> showTranslateDescriptionUI()
+            FEED_CARD_SUGGESTED_EDITS_IMAGE_CAPTION -> showAddImageCaptionUI()
+            FEED_CARD_SUGGESTED_EDITS_TRANSLATE_IMAGE_CAPTION -> showTranslateImageCaptionUI()
+            else -> showAddDescriptionUI()
         }
-        viewArticleTitle.text = sourceSummary!!.normalizedTitle
-        callToActionText.text = if (isTranslation) context.getString(R.string.suggested_edits_feed_card_add_translation_in_language_button, app.language().getAppLanguageCanonicalName(targetSummary!!.lang)) else context.getString(R.string.suggested_edits_add_description_button)
+    }
+
+    private fun showAddDescriptionUI() {
+        viewArticleTitle.text = card!!.sourceSummary!!.normalizedTitle
+        callToActionText.text = if (card!!.invokeSource == FEED_CARD_SUGGESTED_EDITS_TRANSLATE_DESC) String.format(context.getString(R.string.suggested_edits_feed_card_add_translation_in_language_button), app.language().getAppLanguageCanonicalName(card!!.targetSummary!!.lang)) else context.getString(R.string.suggested_edits_add_description_button)
```

Comment:
Should not use `suggested_edits_add_description_button` since it shows only "Add description" instead of "Add article description".

Please add a string resource `R.string.suggested_edits_feed_card_add_description_button` with value "Add article description"